src/engine/LoadURL.cpp

```diff
@@ -0,0 +1,147 @@
+  // query planning time, so we just return `100'000`.
+  return 100'000;
+}
+
+// _____________________________________________________________________________
+float LoadURL::getMultiplicity(size_t) {
+  // TODO: For now, we don't have any information about the multiplicities at
+  // query planning time, so we just return `1` for each column.
+  return 1;
+}
+
+// _____________________________________________________________________________
+bool LoadURL::knownEmptyResult() { return false; }
+
+// _____________________________________________________________________________
+std::unique_ptr<Operation> LoadURL::cloneImpl() const {
+  auto load = std::make_unique<LoadURL>(_executionContext, loadURLClause_);
+  load->cacheBreaker_ = cacheBreaker_;
+  return load;
+}
+
+// _____________________________________________________________________________
+vector<ColumnIndex> LoadURL::resultSortedOn() const { return {}; }
+
+// _____________________________________________________________________________
+Result LoadURL::computeResult(bool) {
+  // TODO<qup42> implement lazy loading; requires modifications to the parser
+  LOG(INFO) << "Loading RDF dataset from " << loadURLClause_.url_.asString()
+            << std::endl;
+  HttpOrHttpsResponse response =
+      getResultFunction_(loadURLClause_.url_, cancellationHandle_,
+                         boost::beast::http::verb::get, "", "", "");
+
+  auto throwErrorWithContext = [this, &response](std::string_view sv) {
+    std::string ctx;
+    ctx.reserve(100);
+    for (const auto& bytes : std::move(response.body_)) {
+      ctx += std::string(reinterpret_cast<const char*>(bytes.data()),
+                         bytes.size());
+      if (ctx.size() >= 100) {
+        break;
+      }
+    }
+    this->throwErrorWithContext(sv, std::string_view(ctx).substr(0, 100));
+  };
+
+  if (response.status_ != boost::beast::http::status::ok) {
+    throwErrorWithContext(absl::StrCat(
+        "RDF dataset responded with HTTP status code: ",
+        static_cast<int>(response.status_), ", ",
+        toStd(boost::beast::http::obsolete_reason(response.status_))));
+  }
+  std::optional<ad_utility::MediaType> mediaType =
+      ad_utility::toMediaType(response.contentType_);
+  if (!mediaType) {
+    throwErrorWithContext(
+        "QLever requires the endpoint of a LoadURL to return the mediatype.");
+  }
+  if (!ad_utility::contains(SUPPORTED_MEDIATYPES, mediaType.value())) {
+    throwErrorWithContext(
+        absl::StrCat("Unsupported media type \"", toString(mediaType.value()),
+                     "\". Supported media types are \"text/turtle\" and "
+                     "\"application/n-triples\"."));
+  }
+  using Re2Parser = RdfStringParser<TurtleParser<Tokenizer>>;
+  auto parser = Re2Parser();
+  std::string body;
+  for (const auto& bytes : response.body_) {
+    body.append(reinterpret_cast<const char*>(bytes.data()), bytes.size());
+  }
+  parser.setInputStream(body);
```

Comment:
We have an `RdfStreamParser` or something that can buffer the reading etc. The thing you probably have to do, is write a class that inherits from the `ParallelBuffer` (currently we only have `readingFromFile` there not `reading from URL/an arbitrary generator`, but that should be not a big problem.
But maybe that can be a separate PR to split up all the infrastructure (smaller functions etc.) and then do the proper thing that is efficient).

Reply:
Planned for seperate PRs (one extending the parser und one using it here): https://github.com/ad-freiburg/qlever-private/issues/49.